src/main/native/smithwaterman/PairWiseSW.h

```diff
@@ -452,8 +459,9 @@ int32_t CONCAT(runSWOnePairBT_,SIMD_ENGINE)(int32_t match, int32_t mismatch, int
     SeqPair p;
     p.seq1 = seq1;
     p.seq2 = seq2;
-    p.len1 = len1;
-    p.len2 = len2;
+    // len1 and len2 are bounded by the length of the input with is at most 16 bit.
+    p.len1 = (int16_t)len1;
```

Comment:
Can we look to change " int32_t len1, int32_t len2" to int16 types to align with the 16 bit Max values reported.

Reply:
Done. I also did a similar thing with the strategy value coming from Java.